drill.cpp

```diff
@@ -383,32 +383,46 @@ bool ExcellonProcessor::millhole(std::ofstream &of, double start_x, double start
     // Find the largest z_step that divides 0 through z_work into
     // evenly sized passes such that each pass is at most
     // cutter->stepsize in depth.
-    unsigned int stepcount = 1;
+    int stepcount = 1;
     if (cutter->do_steps) {
-        stepcount = (unsigned int) ceil(abs(cutter->zwork / cutter->stepsize));
+        stepcount = (int) ceil(abs(cutter->zwork / cutter->stepsize));
     }
 
+    double delta_x = stop_x - start_x;
+    double delta_y = stop_y - start_y;
+    double distance = sqrt(delta_x*delta_x + delta_y*delta_y);
     if (cutdiameter * 1.001 >= holediameter) { //In order to avoid a "zero radius arc" error
         // Hole is smaller than cutdiameter so just drill/zig-zag.
         of << "G0 X" << start_x * cfactor << " Y" << start_y * cfactor << '\n';
-        if (slot)
-        {
-            for (unsigned int current_step = 0; true;)
-            {
-                double z = double(current_step+1)/(stepcount) * cutter->zwork;
-                of << "G1 Z" << z * cfactor << '\n';
-                of << "G1 X" << stop_x * cfactor << " Y" << stop_y * cfactor << '\n';
-                current_step++;
-                if (current_step >= stepcount) {
-                    break;
+        if (slot) {
+            // Start one step above Z0 for optimal entry
+            of << "G1 Z" << -1.0/stepcount * cutter->zwork * cfactor
+              << " F" << cutter->vertfeed * cfactor << '\n';
+
+            // Is there enough room for material evacuation?
+            if (distance > 0.3 * cutdiameter) {
+                of  << "F" << cutter->feed * cfactor << '\n';
```

Comment:
Two feedrate lines in a row is not as pretty in the output.  Why not just combine this output and the one above it into if-else?

Reply:
I don't understand. This is not on the same line.

This comes out as
```
G0 X3.81500 Y-2.66535
G1 Z0.02100 F50.00000
F100.00000
G1 X3.81500 Y-2.58665 Z0.01050
```

So it does the initial plunge with F50, then changes the feedrate to F100 for the rest of the moves if the hole/slot is big enough for the horizontal feedrate).

Reply:
Ah, okay, now I follow!  Right, that's makes sense.  I forgot that the F50 actually affects the text on the same line, too, so it is actually setting the feedrate for that G1 and then setting it again.  Cool.